arch/arm/boot/dts/overlays/pcie-32bit-dma-overlay-pi5.dts

```diff
@@ -0,0 +1,26 @@
+/*
+ * pcie-32bit-dma-overlay-pi5.dts
+ */
+
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2712";
+
+	fragment@0 {
+		target = <&pcie1>;
+		__overlay__ {
+			/*
+			 * The size of the range is rounded up to a power of 2,
+			 * so the range ends up being 0-4GB, and the MSI vector
+			 * gets pushed beyond 4GB.
+			 */
+			#address-cells = <3>;
+			#size-cells = <2>;
+			dma-ranges = <0x03000000 0x0 0x00000000 0x0 0x00000000
```

Comment:
Why not `dma-range  = <0x02000000 ...`?

Reply:
I am actually unsure as to the function of that first value, so I I left 0x03000000 unchanged from the original bcm2712.dtsi file. If someone more knowledgeable can chime in I would appreciate it 🙂

Reply:
That digit specifies whether it is a 32-bit space or 64-bit space. I'm not sure what effect being given a PCIe address in a 64-bit space would have when running with a 32-bit device, but I'm inclined to say that if it works with 0x02000000 then that would be the better choice. Any thoughts, @P33M?

Reply:
Alright, thanks for clarifying that for me. If that's the case, I think 0x02000000 makes more sense.
I have just made that change, in case that ends up being your decision as well. Let me know if you need me to revert it back to 0x03000000.

Reply:
It's more an issue where multiple ranges are present - there's an order of preference matching prefetchable/64bit/32bit to devices that need each.

Reply:
OK, thanks - I think using 32-bit here is less surprising.